### ISSUE TITLE
Complete ES6+ / ES7 / edge complexity reporting support

### ISSUE THREAD

**Issue by typhonrt** [es-analysis/plato#193]
Greets folks... @jsoverson I have thoroughly rewritten escomplex and have released it under a new NPM module [typhonjs-escomplex](https://www.npmjs.com/package/typhonjs-escomplex). My effort fully supports all next-generation JS syntax via using [babylon](https://github.com/babel/babylon) as the AST generator. I have taken the time to update Plato with PR https://github.com/es-analysis/plato/pull/192. 

Plato has been updated with mixed ES5 / ES6 tests, automatic detection for ES Modules / ES6 for JSHint options, normalized all of the accessing code and HTML templates removing previous shims for complexity data.

Of course I haven't made modifications to Plato to specifically show class level data which is captured with `typhonjs-escomplex`, but all module / class methods of an ES6 / ES Module are represented in Plato reports. Future work on Plato could easily support breaking out reports by class and class methods where applicable. 

To check out the updates for now you can either run:
`npm install git+https://git@github.com/typhonjs-node-escomplex/plato.git`

Or add to `package.json`:
```
dependencies: { 
  "plato": "git+https://git@github.com/typhonjs-node-escomplex/plato.git"
}
```

One can also run plato tests if checking out [my plato fork](https://github.com/typhonjs-node-escomplex/plato) directly after `npm install` execute `npm run test` which will show that all tests pass and in `./tmp` an example report which contains 2 ES5 files and 2 ES6 files (on the right) will display. 

It should be noted that while `typhonjs-escomplex` supports ES7 / edge JS code that other aspects of Plato may not like `jshint` which currently only seems to support ES6. A [blog post](http://jshint.com/blog/new-lang-features/) describes jshint support for new language features.

@jsoverson feel free to let me know if there are any questions.

For interest the entire escomplex rewrite is available here for review: [typhonjs-node-escomplex](https://github.com/typhonjs-node-escomplex). Everything about escomplex has been updated / reviewed / improved with a modular architecture that features plugin support for further extension. Extensive external documentation is forthcoming, but the part that needed to be hooked up to plato is finalized / stable.

**Comment by jsoverson** [comment]
merged #192

**Comment by typhonrt** [comment]
:100: I'll post another PR when the next significant typhonjs-escomplex release is available.

**Comment by ColCh** [comment]
For someone on the internet, how to repair broken build because of github project removal

Fork link is now 404: https://github.com/typhonjs-node-escomplex/plato

--------------

<details><summary>Log from build server</summary>

```
$ yarn
yarn install v0.24.6
[1/4] Resolving packages...
[2/4] Fetching packages...
error Command failed.
Exit code: 128
Command: git
Arguments: clone https://github.com/typhonjs-node-escomplex/plato.git /Users/admin/Library/Caches/Yarn/v1/.tmp/2ab8bb79a15775c382a85c559271d62d
Directory: /Users/admin/FOOOOO
Output:
Cloning into '/Users/admin/Library/Caches/Yarn/v1/.tmp/2ab8bb79a15775c382a85c559271d62d'...
fatal: could not read Username for 'https://github.com': Device not configured
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
ERROR: Job failed: exit status 1
```

</details>

---------------

Just switch to `plato` (`npm i -S plato` or `yarn add plato`) and it should work

git diff for it:

```diff
diff --git a/package.json b/package.json
index 9c9c0c6..999f4a5 100644
--- a/package.json
+++ b/package.json
@@ -65,7 +65,7 @@
     "junk": "^2.1.0",
     "lodash.shuffle": "^4.2.0",
     "memoize-weak": "^1.0.0",
-    "plato": "git+https://github.com/typhonjs-node-escomplex/plato.git#f03c710",
+    "plato": "^1.7.0",
     "promise": "^7.1.1",
     "react": "15.4.2",
     "react-addons-test-utils": "15.4.2",
```

**Comment by typhonrt** [comment]
Things were merged into the main Plato v1.7.0 release back in August, so just use that and it should work for ES6+. 

There will be another major update to the underlying typhonjs-escomplex module later in the year.
